CONTRIBUTING

```diff
@@ -1,36 +1,91 @@
 
 ## Project description
 
 Eclipse Sirius enables the specification of a modeling workbench in terms of
 graphical, table or tree editors with validation rules and actions using
 declarative descriptions.
 
-* http://www.eclipse.org/sirius
-* https://projects.eclipse.org/projects/modeling.sirius
-* http://wiki.eclipse.org/Sirius
+- http://www.eclipse.org/sirius
+- https://projects.eclipse.org/projects/modeling.sirius
+- http://wiki.eclipse.org/Sirius
+
+## Planning
+
+We are working in iteration of 8 weeks.
+The first six weeks are used to improve the product with new features and bug fixes.
+After that, we have a period of two weeks during which we work on non-technical details of the release (documentation, communication, etc) and some technical debts.
+
+## Git
+
+Development is done on the branch master, and all changes must first go through a pull request.
+Obviously, contributors are free to do whatever they want on their local clone(s) and Github forks, the rules above only apply to commits and branches pushed into the official repository.
+
+The work in progress from the official contributors can be visible on their dedicated branches.
+These branches are named using the following naming convention:
+
+```
+three_letters_identifying_the_contributor/kind_of_work/description
+```
+
+You could thus find in our repository branches such as:
+
+- `sbe/fix/diagram-related-issue`
+- `pcd/enh/new-feature-the-the-view-dsl`
+- `gco/doc/architectural-documentation`
+
+We are working with a linear history and all pull requests are rebased on top of the master branch in order to be integrated.
+Contributors are free to create as many commits as they want in the branch of their pull requests, the contributions will be squashed to one or two commits when it will be integrated.
+
+## Commit messages formatting
+
+We are using the following format for all our Git messages:
+
+```
+[XXXX] Short description
+
+Longer description if needed, explaining the reasons of the change and its
+impact, not paraphrasing the patch. The description should use wrapped line.
+
+Bug: https://github.com/eclipse-sirius/sirius-components/issues/XXXX
+Signed-off-by: Author name <author email>
+```
+
+You should always mention the numeric bug id at the start of the first line (e.g. [1247]).
+For the few cases where a commit is not directly related to an issue, use a tag like [releng], [doc] or [cleanup]...
+Look for previously used tag before creating a new one.
+
+## Changelog
+
+All contributions should be mentionned in the file `CHANGELOG.adoc`.
+If a long description is necessary to provide additional details for a commit, it is necessary to provide those details in the changelog.
+Most of our commit messages do not contain a long description since those details are written in the changelog.
+
+Contributions which have a visual impact should try to provide screenshots showing the changes.
```

Comment:
These images referenced from the changelog should be added in `doc/images`, not in the top-level directory.

Reply:
Done